python/pyarrow/_dataset.pyx

```diff
@@ -2083,6 +2090,15 @@ cdef class DirectoryPartitioning(Partitioning):
         return PartitioningFactory.wrap(
             CDirectoryPartitioning.MakeFactory(c_field_names, c_options))
 
+    @property
+    def dictionaries(self):
```

Comment:
Perhaps add a docstring / comment here explaining that this property might be empty or null if the partitioning was not created through some kind of discovery process?  For example:

```
# partitioning.dictionaries will be empty here
partitioning = HivePartitioning(pa.schema([("year", pa.int16()), ("month", pa.int8())]))
```

Reply:
Yes, added some more docstrings now